config/fi_check_package.m4

```diff
@@ -130,7 +130,7 @@ AC_DEFUN([_FI_CHECK_PACKAGE_LIB], [
                           unset fi_Lib])])])])
 
     AS_IF([test "$fi_check_package_lib_happy" = "yes"],
-          [$1_LIBS="-l$2 $4"
+          [$1_LIBS="-l$2 $4 $$1_LIBS"
```

Comment:
It's not just $1_LIBS that is assigned -- it's $1_LDFLAGS, too.

It feels weird for you to call this macro more than once with the same prefix.